cmd/tendermint/commands/rollback_test.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/tendermint/tendermint/internal/state"
+	"github.com/tendermint/tendermint/internal/state/mocks"
+	"github.com/tendermint/tendermint/internal/test/factory"
+	"github.com/tendermint/tendermint/types"
+	"github.com/tendermint/tendermint/version"
+)
+
+func TestRollback(t *testing.T) {
```

Comment:
is there a non-happy path test that we could write?

Reply:
Sure I'll try think of a few. I think the main non-happy paths are a matter of failure to get data i.e. blocks, state or validator sets. I don't think there's too much of a problem with the atomicity of operations (the write only happens once at the end).